models/users.js

```diff
@@ -397,37 +407,150 @@ const addForVerification = async (userId, discordId, profileImageUrl, discordIma
  * @param userId {String} - RDS user Id
  * @return {Promise<{Object}|{Object}>}
  * @throws {Error} - If error occurs while fetching user's image verification entry
  */
-const getUserImageForVerification = async (userId) => {
+const getUserPhotoVerificationRequests = async (userId) => {
   try {
-    const verificationImagesSnapshot = await photoVerificationModel.where("userId", "==", userId).get();
+    const verificationImagesSnapshotQuery = photoVerificationModel.where(
+      "status",
+      "==",
+      photoVerificationRequestStatus.PENDING
+    );
+
+    const user = await userModel.doc(userId).get();
+    if (!user.exists) {
+      throw new Error(`No document with userId: ${userId} was found!`);
+    }
+    const userData = user.data();
+    verificationImagesSnapshotQuery.where("userId", "==", userId);
+    const verificationImagesSnapshot = await verificationImagesSnapshotQuery.get();
     if (verificationImagesSnapshot.empty) {
       throw new Error(`No document with userId: ${userId} was found!`);
     }
-    return verificationImagesSnapshot.docs[0].data();
+
+    const photoVerificationObject = [
+      {
+        ...verificationImagesSnapshot.docs[0].data(),
+        user: {
+          username: userData.username,
+          picture: userData.picture.url,
+        },
+      },
+    ];
+    return photoVerificationObject;
+  } catch (err) {
+    logger.error("Error while Querying Photo Verification Entry", err);
+    throw err;
+  }
+};
+
+/**
+ * Returns all users image verification requests (PENDING only), optionally filtered by username
+ * @param username {String} - RDS username
+ * @return {Promise<{Object}|{Object}>}
+ * @throws {Error} - If error occurs while fetching user's image verification entry
+ */
+const getAllUsersPhotoVerificationRequests = async (username = null) => {
+  try {
+    const verificationImagesSnapshotQuery = photoVerificationModel.where(
+      "status",
+      "==",
+      photoVerificationRequestStatus.PENDING
+    );
+
+    if (username && username !== "") {
+      const user = await userModel.where("username", "==", username).limit(1).get();
+      if (user.empty) {
+        logger.error(`No document with username: ${username} was found!`);
+        return [];
+      }
+      const userData = { id: user.docs[0].id, ...user.docs[0].data() };
+      verificationImagesSnapshotQuery.where("userId", "==", userData.id);
+      const verificationImagesSnapshot = await verificationImagesSnapshotQuery.get();
+      if (verificationImagesSnapshot.empty) {
+        logger.error(`No document with username: ${username} was found!`);
+        return [];
+      }
+
+      const photoVerificationObject = [
+        {
+          ...verificationImagesSnapshot.docs[0].data(),
+          user: {
+            username: userData.username,
+            picture: userData.picture.url,
+          },
+        },
+      ];
+      return photoVerificationObject;
+    }
```

Comment:
Can you please share the model documentation for it?

Reply:
https://github.com/Real-Dev-Squad/website-backend/issues/1094